docs/md/installation.md

```diff
@@ -108,12 +109,111 @@ RewriteRule ^ index.php [L]
 ```
 
 #### Nginx
-If you are using Nginx, the following directive in your site configuration will direct all requests to the `index.php` front controller:
+Use the following configuration (typically placed under `/etc/nginx/sites-available`) for Nginx.
 
 ```
-location / {
-    try_files $uri $uri/ /index.php?$query_string;
+# Example Lychee 4 configuration server block for Nginx. To use, search for (1), (2), and (3) and alter these sections
+# to match your configuration. With the exception of the optional settings at the bottom of this file, nothing else
+# needs to be modified.
+
+server {
+    #Required: Nginx listen and server_name configuration. If TLS support is required, use LetsEncrypt certbot, which
+    #(1)       will acquire certificates and make the required configuration changes to support TLS automatically.
+    
+    listen 80;
+    server_name <mydomain>.<tld>;
+    
+    #Required: root path of Lychee public/ directory. This is the directory that contains index.php and a only few
+    #(2)       subdirectories, not the git root directory that contains readme.md (etc).
+    
+    root /var/www/lychee/public/;
+    
+    #Optional, but highly recommended: ensure directory listings are disabled. This is the Nginx default, but better
+    #                                  safe than sorry if the default has been changed.
+    autoindex off;
+    
+    #Required: the front-end expects to find icons and other graphical resources in /Lychee-front/images/ but the
+    #          git repo path structure places these resources in /img/. Rewrite requests to the correct path.
+    #          !!! Do not modify unless you know what you're doing. !!!
+
+    location  /Lychee-front/images/ { rewrite ^/Lychee-front/images/(.*)$ /img/$1 last; }
+
+    #Required: the front end sends some requests to the the back-end via POST requests sent to /php/index.php,
+    #          but this file is actually /index.php. Rewrite requests to the correct path.
+    #          !!! Do not modify unless you know what you're doing. !!!
+    
+    location = /php/index.php { rewrite ^/(.*)$ /index.php?/$1 last; }
+
+    #Required: the front end sends most requests to the back end via POST requests sent to endpoints in /api.
+    #          Rewrite these requests so they reach /index.php
+    #          !!! Do not modify unless you know what you're doing. !!!
+    
+    location ^~ /api/ { rewrite ^(.*)$ /index.php?/$1 last; }
+
+    #Required: Serve requests for / to index.php (via index directive), otherwise serve files with the specified
+    #          extensions directly. Return 404 for all other file extensions to prevent any erroneously placed
+    #          files from leaking.
+    #          !!! Do not modify unless you know what you're doing. !!!
+    
+    location ~* ^\/$|\.(jpg|jpeg|gif|css|png|js|ico|html|txt|svg)$ { index index.php; }
```

Comment:
What about `.mp4` `.mp3` and other formats such as `.dng`? What about raw formats such as `.nef` or `.cr2` or `.cr3` ?  This is extremely conservative and way too restrictive.

I am not keen into having a white list and would rather see a black list. Yes this is less safe, but those settings are in my opinion way too restrictive.

Additionally, Lychee provides a clean looking 404 page.

Reply:
Thanks for your feedback.

This filter is case insensitive. Welcome to the weird and wonderful world of nginx location directives. :)

The extensions were chosen based on my interpretation of the design intent of the existing nginx recommended configuration, as provided in upgrade.md, and some testing of what filetypes Lychee will accept. The existing nginx configuration directly serves jpg|jpeg|gif|css|png|js|ico|html but passes everything else through a series of `if` and `try_files` blocks that mostly connect to the index.php endpoint. Nginx does not make the behavior of `if` and `try_files` easy to reason about, but my interpretation is that the extension list was intended to be a whitelist.

With the exception of .mp4, the allowed extension list matches formats my installation of Lychee will accept. The omission of mp4 was an oversight. My installation of Lychee will not accept any of the raw formats I have on hand or mp3 audio. Indeed, it will only accept jpeg, gif, png, and mp4.

I feel that, with such a narrow scope of accepted file types, a default-deny policy is adequate.